baseclasses/chemical_energy/neccelectrode.py

```diff
@@ -0,0 +1,189 @@
+        unit=('°C'),
+        a_eln=dict(component='NumberEditQuantity', defaultDisplayUnit='°C'))
+
+    n2_deposition_pressure = Quantity(
+        type=np.dtype(np.float64),
+        unit=('bar'),
+        a_eln=dict(component='NumberEditQuantity', defaultDisplayUnit='bar'))
+
+    mass_loading = Quantity(
+        type=np.dtype(np.float64),
+        unit=('mg/cm^2'),
+        a_eln=dict(component='NumberEditQuantity', defaultDisplayUnit='mg/cm^2'))
+
+    description = Quantity(
+        type=str,
+        description='Any information that cannot be captured in the other fields.',
+        a_eln=dict(
+            component='RichTextEditQuantity',
+            label='Remarks'))
+
+    # TODO check if substrate_type and substrate_dimension is all that necc group wants (right now one empty field in excel)
+    substrate = SubSection(
+        section_def=SubstrateProperties)
+
+    solvents_and_ionomer = SubSection(
+        section_def=CatalystSynthesis, repeats=True)
+
+    def normalize(self, archive, logger):
+        self.chemical_composition_or_formulas = self.electrode_recipe_id.element
+        super(CENECCElectrodeRecipe, self).normalize(archive, logger)
+        export_lab_id(archive, self.lab_id)
+
+
+class CENECCElectrodeID(ReadableIdentifiersCustom):
+
+    m_def = Section(
+        a_eln=dict(
+            hide=["sample_owner", "sample_short_name", "sample_id", "short_name"]
+        ))
+
+    institute = Quantity(
+        type=str,
+        description='Alias/short name of the home institute of the owner, i.e. *HZB*.',
+        default='CE-NECC',
+        a_eln=dict(
+            component='EnumEditQuantity',
+            props=dict(
+                suggestions=[
+                    'CE-NECC'])))
+
+    owner = Quantity(
+        type=str,
+        default='Matthew Mayer',
+        a_eln=dict(
+            component='EnumEditQuantity',
+            props=dict(
+                suggestions=[
+                    'Matthew Mayer',
+                    'Siddharth Gupta',
+                    'Christina Roukounaki',
+                    'Gumaa El-Nagar',
+                    'Uttam Gupta',
+                ])))
+
+    recipe = Quantity(
+        type=Reference(CENECCElectrodeRecipe.m_def),
+        a_eln=dict(component='ReferenceEditQuantity'))
+
+    def normalize(self, archive, logger):
+        super(CENECCElectrodeID, self).normalize(archive, logger)
+
+        if archive.data.lab_id:
+            return
+        if not self.datetime:
+            from datetime import date
+            self.datetime = date.today()
+
+        if self.institute and self.owner and self.datetime:
+            id_base = [build_initial_id(self.institute, self.owner, self.datetime),
+                       self.recipe.lab_id]
+            self.lab_id = '_'.join(id_base)
+        create_id(archive, self.lab_id)
+
+class CENECCElectrode(CESample):
+    # TODO class name maybe CENECCSample or something with 'catalyst detail'
```

Comment:
ID creation with Electrode and ElectrodeRecipe. Is this solved correctly on the NOMAD side? I have merged the separation of cathode and anode from the excel table into one class. Is the naming from the chemical side still correct?

Reply:
not sure what you mean